src/messaging/ExchangeContext.cpp

```diff
@@ -531,58 +532,65 @@ CHIP_ERROR ExchangeContext::HandleMessage(uint32_t messageCounter, const Payload
         MessageHandled();
     });
 
-    if (mDispatch.IsReliableTransmissionAllowed() && !IsGroupExchangeContext())
+    if (GetSessionHandle()->AllowsMRP())
```

Comment:
Is there a reason this is not just `if (mSession->AllowsMRP())`?  Seems like it should be the same thing in the end, but clearer...  Same for the condition below.

Reply:
No particular reason. We could use `mSession` as well. Will change.